runtime/compiler/env/j9methodServer.cpp

```diff
@@ -633,62 +612,31 @@ TR_ResolvedJ9JITServerMethod::getResolvedStaticMethod(TR::Compilation * comp, I_
 
    auto compInfoPT = (TR::CompilationInfoPerThreadRemote *) _fe->_compInfoPT;
    TR_ResolvedMethod *resolvedMethod = NULL;
-   if (compInfoPT->getCachedResolvedMethod(compInfoPT->getResolvedMethodKey(TR_ResolvedMethodType::Static, (TR_OpaqueClassBlock *) _ramClass, cpIndex), this, &resolvedMethod, unresolvedInCP))
+   TR_ResolvedMethodKey key = compInfoPT->getResolvedMethodKey(TR_ResolvedMethodType::Static, (TR_OpaqueClassBlock *) _ramClass, cpIndex);
+   if (compInfoPT->getCachedResolvedMethod(key, this, &resolvedMethod, unresolvedInCP))
       {
       if ((resolvedMethod == NULL) && unresolvedInCP)
          handleUnresolvedStaticMethodInCP(cpIndex, unresolvedInCP);
       return resolvedMethod;
       }
 
    _stream->write(JITServer::MessageType::ResolvedMethod_getResolvedStaticMethodAndMirror, _remoteMirror, cpIndex);
-   auto recv = _stream->read<J9Method *, TR_ResolvedJ9JITServerMethodInfo, bool>();
-   J9Method * ramMethod = std::get<0>(recv);
+   auto recv = _stream->read<TR_OpaqueMethodBlock *, TR_ResolvedJ9JITServerMethodInfo, bool>();
+   TR_OpaqueMethodBlock *ramMethod = std::get<0>(recv);
+   auto methodInfo = std::get<1>(recv);
    bool isUnresolvedInCP = std::get<2>(recv);
    if (unresolvedInCP)
       *unresolvedInCP = isUnresolvedInCP;
 
-   if (comp->compileRelocatableCode() && comp->getOption(TR_UseSymbolValidationManager) && ramMethod)
+   if (ramMethod == NULL) 
       {
-      if (!comp->getSymbolValidationManager()->addStaticMethodFromCPRecord((TR_OpaqueMethodBlock *)ramMethod, cp(), cpIndex))
-         ramMethod = NULL;
-      }
-
-   bool skipForDebugging = doResolveAtRuntime(ramMethod, cpIndex, comp);
-   if (isArchetypeSpecimen())
-      {
-      // ILGen macros currently must be resolved for correctness, or else they
-      // are not recognized and expanded.  If we have unresolved calls, we can't
-      // tell whether they're ilgen macros because the recognized-method system
-      // only works on resovled methods.
-      //
-      if (ramMethod)
-         skipForDebugging = false;
-      else
+      if (unresolvedInCP) 
          {
-         comp->failCompilation<TR::ILGenFailure>("Can't compile an archetype specimen with unresolved calls");
-         }
-      }
```

Comment:
This code is no longer executed. Is this correct?

Reply:
This code will just run on the client instead, so if needed, the compilation will be failed there, instead of on the server.